rpcs3/util/sysinfo.cpp

```diff
@@ -68,14 +68,76 @@ namespace Darwin_ProcessInfo
 }
 #endif
 
-#if defined(ARCH_ARM64) && defined (_WIN32) && !defined(_MSC_VER)
-static inline u64 __readx18qword(int offset)
+namespace utils
 {
-	u64* peb = nullptr;
-	__asm__ volatile("mov %0, x18" : "=r"(peb));
-	return peb[offset];
-}
+#ifdef _WIN32
+	// Alternative way to read OS version using the registry.
+	static std::string get_fallback_windows_version()
+	{
+		// Some helpers for sanity
+		const auto read_reg_dword = [](HKEY hKey, std::string_view value_name) -> std::pair<bool, DWORD>
+		{
+			DWORD val;
+			DWORD len = sizeof(val);
+			if (ERROR_SUCCESS != RegQueryValueExA(hKey, value_name.data(), nullptr, nullptr, reinterpret_cast<LPBYTE>(&val), &len))
+			{
+				return { false, 0 };
+			}
+			return { true, val };
+		};
+
+		const auto read_reg_sz = [](HKEY hKey, std::string_view value_name) -> std::pair<bool, std::string>
+		{
+			char sz[256];
```

Comment:
don't you need 257 characters? so at index 256 there always be a null terminator?

Reply:
Makes more sense to just set max-len to 255

Reply:
Done.